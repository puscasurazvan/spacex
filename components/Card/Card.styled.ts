import styled from 'styled-components';

export const CardWrapper = styled.div({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  boxShadow: '0 0 10px 0 rgba(0,0,0,0.45)',
  margin: '10px',
  borderRadius: '15px',
  padding: '10px',
  width: '100%',
  fontFamily: 'Orbitron Regular',
  [`@media (min-width: 992px)`]: {
    width: '500px',
  },
  '&:hover': {
    boxShadow:
      '0px -25px 20px -20px rgba(0, 0, 0, 0.45), 25px 0 20px -20px rgba(0, 0, 0, 0.45), 0px 25px 20px -20px rgba(0, 0, 0, 0.45), -25px 0 20px -20px rgba(0, 0, 0, 0.45)',
    borderRadius: '25px',

    transition: 'box-shadow 500ms, border-radius 1000ms',
  },
});

export const Heading = styled.h2({
  textAlign: 'center',
  textShadow: '1px 1px #8e8e8e',
  borderBottom: '1px solid lightgrey',
  width: '100%',
  paddingBottom: '20px',
});

export const ImageWrapper = styled.div({
  margin: '0 auto',
});
