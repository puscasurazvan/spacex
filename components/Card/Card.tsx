import * as Styled from './Card.styled';

import Image from 'next/image';

import { Badge } from '../Badge';
import { LaunchDetails } from '../LaunchDetails';

type CardProps = {
  imageSrc: string;
  status: string;
  name: string;
  date: string;
  core: string;
  payloadType: string;
  payloadId: string;
  failureReason: string;
};

export const Card = ({ imageSrc, status, name, date, core, payloadType, payloadId, failureReason }: CardProps) => {
  return (
    <Styled.CardWrapper>
      <Styled.Heading data-testid="heading">
        {name} {status === 'Success' ? 'ğŸš€' : 'ğŸ’¥'}
      </Styled.Heading>

      <Styled.ImageWrapper>
        <Image src={imageSrc} width="100" height="100" alt="SpaceX Image" />
      </Styled.ImageWrapper>
      <Badge launchStatus={status} />

      <LaunchDetails
        launchDate={date}
        core={core}
        payloadType={payloadType}
        payloadId={payloadId}
        failureReason={failureReason}
      />
    </Styled.CardWrapper>
  );
};
