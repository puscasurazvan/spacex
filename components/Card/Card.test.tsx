import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Card } from './Card';

describe('Card Component', () => {
  const imageSrc = 'https://images2.imgbox.com/94/f2/NN6Ph45r_o.png';
  const status = 'Success';
  const name = 'Falcon 9 Launch';
  const date = '2023-07-27';
  const core = 'B1058';
  const payloadType = 'Satellite';
  const payloadId = 'ABCD1234';
  const failureReason = '';
  const props = { imageSrc, status, name, date, core, payloadType, payloadId, failureReason };

  it('snapshot testing', () => {
    const { container } = render(<Card {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the Card component with all details and success status', () => {
    const { getByAltText, getByTestId } = render(<Card {...props} />);
    const heading = getByTestId('heading');
    const imageElement = getByAltText('SpaceX Image');
    const badgeElement = getByTestId('badge');

    expect(heading).toBeInTheDocument();
    expect(imageElement).toBeInTheDocument();
    expect(badgeElement).toBeInTheDocument();
    expect(heading).toHaveTextContent('ðŸš€');
  });

  it('renders the Card component with failure status and failure reason', () => {
    const failureReason = 'Engine anomaly detected';
    const status = 'Failure';
    const { getByText, getByAltText, getByTestId } = render(
      <Card
        imageSrc={imageSrc}
        status={status}
        name={name}
        date={date}
        core={core}
        payloadType={payloadType}
        payloadId={payloadId}
        failureReason={failureReason}
      />
    );

    const heading = getByTestId('heading');
    const imageElement = getByAltText('SpaceX Image');
    const badgeElement = getByTestId('badge');

    expect(heading).toBeInTheDocument();
    expect(imageElement).toBeInTheDocument();
    expect(badgeElement).toBeInTheDocument();

    const failureReasonElement = getByText(failureReason);
    expect(failureReasonElement).toBeInTheDocument();
    expect(heading).toHaveTextContent('ðŸ’¥');
  });
});
