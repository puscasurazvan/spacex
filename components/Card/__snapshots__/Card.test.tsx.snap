// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component snapshot testing 1`] = `
<div>
  <div
    class="sc-bczSft DvdJH"
  >
    <h2
      class="sc-gsnScm jUQJBz"
      data-testid="heading"
    >
      Falcon 9 Launch
       
      ðŸš€
    </h2>
    <div
      class="sc-dkzCjb JynFz"
    >
      <img
        alt="SpaceX Image"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimages2.imgbox.com%2F94%2Ff2%2FNN6Ph45r_o.png&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
    </div>
    <h3
      class="sc-hKMtDE jCeMsk"
      color="green"
      data-testid="badge"
    >
      Success
    </h3>
    <ul
      class="sc-eCYdKt cmpqUZ"
    >
      <li
        class="sc-jSMdHm fuUGvV"
        data-testid="date"
      >
        <span
          class="sc-gKXNOb bsSupN"
        >
          DATE:
        </span>
         
        2023-07-27
      </li>
      <li
        class="sc-jSMdHm fuUGvV"
        data-testid="core"
      >
        <span
          class="sc-gKXNOb bsSupN"
        >
          CORE:
        </span>
         
        B1058
      </li>
      <li
        class="sc-jSMdHm fuUGvV"
        data-testid="payloadType"
      >
        <span
          class="sc-gKXNOb bsSupN"
        >
          PAYLOAD TYPE:
        </span>
         
        Satellite
      </li>
      <li
        class="sc-jSMdHm fuUGvV"
        data-testid="payloadId"
      >
        <span
          class="sc-gKXNOb bsSupN"
        >
          PAYLOAD ID:
        </span>
         
        ABCD1234
      </li>
    </ul>
  </div>
</div>
`;
